### ISSUE TITLE
rqlite panics from nil deref on exit

### ISSUE THREAD

**Issue by jtackaberry** [rqlite/rqlite#1486]
**What version are you running?**

8.11.0

**Are you using Docker or Kubernetes to run your system?**

Yes (Microk8s) with the official image (`docker.io/rqlite/rqlite:8.11.0`)

**Are you running a single node or a cluster?**

3 node cluster.

**What did you do?**

Exec'd into one of the pod nodes, ran `rqlite` and simply exited (either `.quit` or ctrl-d):

```
~ $ rqlite
Welcome to the rqlite CLI.
Enter ".help" for usage hints.
Connected to http://127.0.0.1:4001 running version v8.11.0
127.0.0.1:4001> .quit
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x6b8061]

goroutine 1 [running]:
main.main.func1(0xc0000a8420)
        /tmp/tmp.Nem43vRKne/src/github.com/rqlite/rqlite/cmd/rqlite/main.go:236 +0x14c1
github.com/mkideal/cli.(*Command).RunWith(0xc0000b0360, {0xc0000240a0, 0x0, 0x0}, {0x0?, 0x0?}, {0x0, 0x0}, {0x0, 0x0, ...})
        /tmp/tmp.Nem43vRKne/pkg/mod/github.com/mkideal/cli@v0.2.7/command.go:219 +0x3c7
github.com/mkideal/cli.(*Command).Run(...)
        /tmp/tmp.Nem43vRKne/pkg/mod/github.com/mkideal/cli@v0.2.7/command.go:176
github.com/mkideal/cli.RunWithArgs({0x6e4ee0?, 0xc0000a8370}, {0xc0000240a0, 0x1, 0x1}, 0x771e88, {0x0?, 0x973048?, 0xc0000061a0?})
        /tmp/tmp.Nem43vRKne/pkg/mod/github.com/mkideal/cli@v0.2.7/cli.go:30 +0x238
github.com/mkideal/cli.Run(...)
        /tmp/tmp.Nem43vRKne/pkg/mod/github.com/mkideal/cli@v0.2.7/cli.go:15
main.main()
        /tmp/tmp.Nem43vRKne/src/github.com/rqlite/rqlite/cmd/rqlite/main.go:73 +0x59
```

**What did you expect to happen?**

Not panic :)

**What happened instead?**

Panic

**Please include the Status, Nodes, and Expvar output from each node (or at least the Leader!)**

From leader:

```
Welcome to the rqlite CLI.
Enter ".help" for usage hints.
Connected to http://127.0.0.1:4001 running version v8.11.0
127.0.0.1:4001> .status
node:
  uptime: 4m43.413110696s
  current_time: 2023-12-17T18:27:46.604845496Z
  start_time: 2023-12-17T18:23:03.191735351Z
runtime:
  GOARCH: amd64
  GOMAXPROCS: 10
  GOOS: linux
  num_cpu: 10
  num_goroutine: 43
  version: go1.21.5
http:
  auth: enabled
  bind_addr: [::]:4001
  cluster:
    conn_pool_stats:
      10.3.90.55:4002:
        open_connections: 4
        idle: 4
        max_open_connections: 64
      rqlite-0.rqlite-headless.data.svc.cluster.local:4002:
        idle: 1
        max_open_connections: 64
        open_connections: 5
      rqlite-2.rqlite-headless.data.svc.cluster.local:4002:
        idle: 4
        max_open_connections: 64
        open_connections: 4
      10.3.50.167:4002:
        idle: 4
        max_open_connections: 64
        open_connections: 4
    local_node_addr: rqlite-1.rqlite-headless.data.svc.cluster.local:4002
    timeout: 30s
  queue:
    _default:
      batch_size: 128
      max_size: 1024
      sequence_number: 0
      timeout: 50ms
  tls:
    enabled: false
network:
  interfaces:
    eth0:
      addresses: [map[address:10.3.90.55/32] map[address:2001:470:b0d6:fd:8a42:53cc:a14a:5a37/128] map[address:fe80::3034:45ff:fe2c:910/64]]
      flags: up|broadcast|multicast|running
      hardware_address: 32:34:45:2c:09:10
    lo:
      flags: up|loopback|running
      hardware_address:
      addresses: [map[address:127.0.0.1/8] map[address:::1/128]]
os:
  page_size: 4096
  pid: 1
  ppid: 0
  executable: /bin/rqlited
  hostname: rqlite-1
store:
  open: true
  addr: rqlite-1.rqlite-headless.data.svc.cluster.local:4002
  db_conf:
    fk_constraints: true
  dir_size: 69632
  election_timeout: 1s
  raft:
    fsm_pending: 0
    last_log_index: 6
    log_size: 32768
    num_peers: 2
    snapshot_version_min: 0
    term: 6
    state: Leader
    bolt:
      FreePageN: 0
      PendingPageN: 2
      FreeAlloc: 8192
      FreelistInuse: 32
      TxN: 78
      OpenTxN: 0
      TxStats:
        Rebalance: 0
        RebalanceTime: 0
        SpillTime: 136390
        WriteTime: 114343410
        PageCount: 19
        PageAlloc: 77824
        NodeCount: 18
        Spill: 9
        Write: 29
        CursorCount: 187
        NodeDeref: 0
        Split: 0
    commit_index: 6
    last_contact: 0
    last_log_term: 6
    last_snapshot_index: 0
    protocol_version: 3
    protocol_version_max: 3
    applied_index: 6
    last_snapshot_term: 0
    latest_configuration: [{Suffrage:Voter ID:rqlite-0 Address:rqlite-0.rqlite-headless.data.svc.cluster.local:4002} {Suffrage:Voter ID:rqlite-1 Address:rqlite-1.rqlite-headless.data.svc.cluster.local:4002} {Suffrage:Voter ID:rqlite-2 Address:rqlite-2.rqlite-headless.data.svc.cluster.local:4002}]
    snapshot_version_max: 1
    voter: true
    latest_configuration_index: 0
    protocol_version_min: 0
  ready: true
  reap_timeout: 0s
  leader:
    addr: rqlite-1.rqlite-headless.data.svc.cluster.local:4002
    node_id: rqlite-1
  nodes: [map[addr:rqlite-0.rqlite-headless.data.svc.cluster.local:4002 id:rqlite-0 suffrage:Voter] map[addr:rqlite-1.rqlite-headless.data.svc.cluster.local:4002 id:rqlite-1 suffrage:Voter] map[addr:rqlite-2.rqlite-headless.data.svc.cluster.local:4002 id:rqlite-2 suffrage:Voter]]
  fsm_index: 0
  node_id: rqlite-1
  apply_timeout: 10s
  heartbeat_timeout: 1s
  request_marshaler:
    compression_batch: 50
    compression_size: 1024
    force_compression: false
  no_freelist_sync: false
  reap_read_only_timeout: 0s
  sqlite3:
    conn_pool_stats:
      ro:
        idle: 1
        wait_count: 0
        max_open_connections: 0
        in_use: 0
        wait_duration: 0
        max_idle_closed: 0
        max_idle_time_closed: 0
        max_lifetime_closed: 0
        open_connections: 1
      rw:
        max_open_connections: 1
        wait_count: 0
        wait_duration: 0
        max_idle_closed: 0
        max_idle_time_closed: 0
        max_lifetime_closed: 0
        open_connections: 1
        in_use: 0
        idle: 1
    db_size: 4096
    mem_stats:
      cache_size: -2000
      freelist_count: 0
      hard_heap_limit: 0
      max_page_count: 1073741823
      page_count: 1
      page_size: 4096
      soft_heap_limit: 0
    rw_dsn: file:/rqlite/db.sqlite?_fk=true
    compile_options: [ATOMIC_INTRINSICS=1 COMPILER=gcc-9.4.0 DEFAULT_AUTOVACUUM DEFAULT_CACHE_SIZE=-2000 DEFAULT_FILE_FORMAT=4 DEFAULT_JOURNAL_SIZE_LIMIT=-1 DEFAULT_MMAP_SIZE=0 DEFAULT_PAGE_SIZE=4096 DEFAULT_PCACHE_INITSZ=20 DEFAULT_RECURSIVE_TRIGGERS DEFAULT_SECTOR_SIZE=4096 DEFAULT_SYNCHRONOUS=2 DEFAULT_WAL_AUTOCHECKPOINT=1000 DEFAULT_WAL_SYNCHRONOUS=1 DEFAULT_WORKER_THREADS=0 ENABLE_DBSTAT_VTAB ENABLE_FTS3 ENABLE_FTS3_PARENTHESIS ENABLE_FTS5 ENABLE_RTREE ENABLE_UPDATE_DELETE_LIMIT MALLOC_SOFT_LIMIT=1024 MAX_ATTACHED=10 MAX_COLUMN=2000 MAX_COMPOUND_SELECT=500 MAX_DEFAULT_PAGE_SIZE=8192 MAX_EXPR_DEPTH=1000 MAX_FUNCTION_ARG=127 MAX_LENGTH=1000000000 MAX_LIKE_PATTERN_LENGTH=50000 MAX_MMAP_SIZE=0x7fff0000 MAX_PAGE_COUNT=1073741823 MAX_PAGE_SIZE=65536 MAX_SQL_LENGTH=1000000000 MAX_TRIGGER_DEPTH=1000 MAX_VARIABLE_NUMBER=32766 MAX_VDBE_OP=250000000 MAX_WORKER_THREADS=8 MUTEX_PTHREADS OMIT_DEPRECATED OMIT_LOAD_EXTENSION OMIT_SHARED_CACHE SYSTEM_MALLOC TEMP_STORE=1 THREADSAFE=1]
    path: /rqlite/db.sqlite
    pragmas:
      ro:
        foreign_keys: 1
        journal_mode: wal
        synchronous: 1
        wal_autocheckpoint: 1000
      rw:
        foreign_keys: 1
        journal_mode: wal
        synchronous: 0
        wal_autocheckpoint: 0
    ro_dsn: file:/rqlite/db.sqlite?mode=ro&_fk=true
    size: 4096
    version: 3.44.0
    wal_size: 0
  trailing_logs: 10240
  last_applied_index: 6
  observer:
    dropped: 0
    observed: 25
  snapshot_store:
    db_path:
    dir: /rqlite/rsnapshots
    snapshots: []
  snapshot_threshold: 8192
  snapshot_interval: 30s
  db_applied_index: 0
  dir: /rqlite
build:
  build_time: 2023-12-17T12:19:32-0500
  commit: aad2042be1204b1e7f290a759c21a387f5566684
  compiler: gc
  version: v8.11.0
  branch: master
cluster:
  addr: rqlite-1.rqlite-headless.data.svc.cluster.local:4002
  api_addr: rqlite-1.rqlite-headless.data.svc.cluster.local:4001
  https: false
disco:
  last_addresses: [10.3.90.55:4002 10.3.50.167:4002 10.3.59.73:4002]
  last_contact: 2023-12-17T18:23:03.254467214Z
  mode: dns
  name: rqlite-headless
  port: 4002
127.0.0.1:4001> .nodes
rqlite-0:
  reachable: true
  leader: false
  time: 6.61e-7
  time_s: 1.613µs
  id: rqlite-0
  api_addr: http://rqlite-0.rqlite-headless.data.svc.cluster.local:4001
  addr: rqlite-0.rqlite-headless.data.svc.cluster.local:4002
  voter: true
rqlite-1:
  api_addr: http://rqlite-1.rqlite-headless.data.svc.cluster.local:4001
  addr: rqlite-1.rqlite-headless.data.svc.cluster.local:4002
  voter: true
  reachable: true
  leader: true
  time: 1e-7
  time_s: 261ns
  id: rqlite-1
rqlite-2:
  leader: false
  time: 4.2e-7
  time_s: 1.042µs
  id: rqlite-2
  api_addr: http://rqlite-2.rqlite-headless.data.svc.cluster.local:4001
  addr: rqlite-2.rqlite-headless.data.svc.cluster.local:4002
  voter: true
  reachable: true
127.0.0.1:4001> .expvar
cmdline: [/bin/rqlited -node-id rqlite-1 -http-addr 0.0.0.0:4001 -http-adv-addr rqlite-1.rqlite-headless.data.svc.cluster.local:4001 -raft-addr 0.0.0.0:4002 -raft-adv-addr rqlite-1.rqlite-headless.data.svc.cluster.local:4002 -auth=/config/users.json -fk=true -disco-mode=dns -disco-config={"name":"rqlite-headless"} -bootstrap-expect=3 -join-as=rqlite -join-interval=1s -join-attempts=120 -raft-shutdown-stepdown /rqlite]
proto:
  num_compressed_bytes: 0
  num_compressed_requests: 0
  num_compression_misses: 0
  num_precompressed_bytes: 0
  num_requests: 0
  num_uncompressed_bytes: 0
  num_uncompressed_requests: 0
queue:
  num_timeout: 0
  statements_rx: 0
  statements_tx: 0
  num_flush: 0
http:
  remote_loads: 0
  remote_requests: 0
  backups: 0
  executions: 0
  num_status: 1
  queued_executions_not_leader: 0
  remote_backups: 0
  remote_executions: 0
  authOK: 61
  boot: 0
  loads_aborted: 0
  queued_executions_no_leader: 0
  queued_executions_num_stmts_rx: 0
  queued_executions_num_stmts_tx: 0
  queued_executions_ok: 0
  remote_executions_failed: 0
  remote_queries: 0
  remote_remove_node: 0
  leader_not_found: 0
  queued_executions: 0
  request_stmts_rx: 0
  authFail: 0
  query_stmts_rx: 0
  queued_executions_failed: 0
  queued_executions_wait: 0
  remote_queries_failed: 0
  requests: 0
  execute_stmts_rx: 0
  queries: 0
  remote_requests_failed: 0
  loads: 0
  num_readyz: 57
  queued_executions_leadership_lost: 0
  queued_executions_unknown_error: 0
memstats:
  Alloc: 6219880
  HeapIdle: 3915776
  StackSys: 1015808
  PauseEnd: [1702837383189639013 1702837383341176239 1702837410115398643 1702837441313716234 1702837488275591731 1702837538268102325 1702837589645750222 1702837638293588651 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]
  Lookups: 0
  NextGC: 8910360
  BySize: [map[Frees:0 Mallocs:0 Size:0] map[Frees:267 Mallocs:1350 Size:8] map[Frees:24825 Mallocs:31170 Size:16] map[Frees:11033 Mallocs:12889 Size:24] map[Frees:16594 Mallocs:18982 Size:32] map[Frees:15599 Mallocs:19335 Size:48] map[Frees:36179 Mallocs:41178 Size:64] map[Frees:35708 Mallocs:40367 Size:80] map[Frees:23114 Mallocs:26668 Size:96] map[Frees:542 Mallocs:654 Size:112] map[Frees:196 Mallocs:328 Size:128] map[Frees:6572 Mallocs:16247 Size:144] map[Frees:69 Mallocs:175 Size:160] map[Frees:4 Mallocs:114 Size:176] map[Frees:56 Mallocs:119 Size:192] map[Frees:89 Mallocs:379 Size:208] map[Frees:137 Mallocs:164 Size:224] map[Frees:18 Mallocs:38 Size:240] map[Frees:68 Mallocs:122 Size:256] map[Frees:59 Mallocs:681 Size:288] map[Frees:65 Mallocs:87 Size:320] map[Frees:160 Mallocs:334 Size:352] map[Frees:6 Mallocs:44 Size:384] map[Frees:20 Mallocs:175 Size:416] map[Frees:1 Mallocs:35 Size:448] map[Frees:5 Mallocs:18 Size:480] map[Frees:6269 Mallocs:7080 Size:512] map[Frees:12 Mallocs:77 Size:576] map[Frees:26 Mallocs:42 Size:640] map[Frees:20 Mallocs:33 Size:704] map[Frees:0 Mallocs:12 Size:768] map[Frees:76 Mallocs:110 Size:896] map[Frees:3 Mallocs:43 Size:1024] map[Frees:6 Mallocs:98 Size:1152] map[Frees:54 Mallocs:88 Size:1280] map[Frees:4 Mallocs:8 Size:1408] map[Frees:0 Mallocs:2 Size:1536] map[Frees:4 Mallocs:21 Size:1792] map[Frees:14 Mallocs:22 Size:2048] map[Frees:3 Mallocs:132 Size:2304] map[Frees:14 Mallocs:19 Size:2688] map[Frees:3 Mallocs:6 Size:3072] map[Frees:0 Mallocs:1 Size:3200] map[Frees:0 Mallocs:3 Size:3456] map[Frees:33 Mallocs:60 Size:4096] map[Frees:14 Mallocs:33 Size:4864] map[Frees:2 Mallocs:10 Size:5376] map[Frees:0 Mallocs:5 Size:6144] map[Frees:0 Mallocs:0 Size:6528] map[Frees:0 Mallocs:0 Size:6784] map[Frees:2 Mallocs:3 Size:6912] map[Frees:2 Mallocs:15 Size:8192] map[Frees:5 Mallocs:16 Size:9472] map[Frees:0 Mallocs:0 Size:9728] map[Frees:0 Mallocs:0 Size:10240] map[Frees:0 Mallocs:0 Size:10880] map[Frees:2 Mallocs:2 Size:12288] map[Frees:0 Mallocs:0 Size:13568] map[Frees:0 Mallocs:0 Size:14336] map[Frees:0 Mallocs:4 Size:16384] map[Frees:1 Mallocs:1 Size:18432]]
  PauseTotalNs: 1055672
  NumGC: 8
  DebugGC: false
  TotalAlloc: 21792008
  Sys: 20550920
  MCacheSys: 15600
  LastGC: 1702837638293588651
  HeapAlloc: 6219880
  MSpanSys: 228144
  MCacheInuse: 12000
  HeapObjects: 41655
  StackInuse: 1015808
  GCSys: 4023064
  OtherSys: 2249501
  Mallocs: 228246
  Frees: 186591
  HeapSys: 11567104
  HeapInuse: 7651328
  PauseNs: [223063 72547 132391 110490 189208 145075 86164 96734 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]
  GCCPUFraction: 0.00001470554838089387
  NumForcedGC: 0
  EnableGC: true
  HeapReleased: 2400256
  MSpanInuse: 203112
  BuckHashSys: 1451699
mux:
  num_connections_handled: 15
  num_unregistered_handlers: 0
snapshot:
  latest_persist_duration: 0
  latest_persist_size: 0
  upgrade_fail: 0
  upgrade_ok: 0
store:
  num_snapshots: 0
  snapshot_precompact_wal_size: 0
  num_removed_before_joins: 0
  num_auto_restores: 0
  num_compressed_commands: 0
  num_snapshots_incremental: 0
  nodes_reaped_ok: 0
  num_restores: 0
  num_boots: 0
  num_ignored_joins: 0
  num_provides: 0
  num_user_snapshots: 0
  snapshot_wal_size: 0
  leader_changes_dropped: 0
  num_auto_restores_failed: 0
  num_recoveries: 0
  num_snapshots_full: 0
  num_auto_restores_skipped: 0
  num_joins: 0
  num_backups: 0
  leader_changes_observed: 0
  nodes_reaped_failed: 0
  num_db_stats_errors: 0
  snapshot_persist_duration: 0
  num_uncompressed_commands: 0
  snapshot_create_duration: 0
  failed_heartbeat_observed: 22
  num_loads: 0
cluster:
  num_load_req: 0
  num_client_retries: 4
  num_get_node_api_req: 106
  num_get_node_api_req_local: 57
  num_get_node_api_resp: 106
  num_query_req: 0
  num_remove_node_req: 0
  num_request_req: 0
  num_backup_req: 0
  num_execute_req: 0
  num_join_req: 3
  num_notify_req: 1
db:
  checkpointed_moves: 0
  checkpointed_pages: 0
  query_transactions: 0
  checkpoint_duration_ns: 0
  checkpoints: 0
  executions: 0
  queries: 9
  query_errors: 0
  request_transactions: 0
  requests: 0
  checkpoint_errors: 0
  execute_transactions: 0
  execution_errors: 0
  open_duration_ms: 13
downloader:
  num_downloads_ok: 0
  download_bytes: 0
  num_downloads_fail: 0
uploader:
  last_upload_bytes: 0
  num_uploads_fail: 0
  num_uploads_ok: 0
  num_uploads_skipped: 0
  total_upload_bytes: 0
```

**Comment by jtackaberry** [comment]
(Updated OP to reflect 8.11.0 which was just released.)

**Comment by otoolep** [comment]
Thanks for the report, fixed in 8.11.1.
